fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,180 @@
+                                   "the plotting for MetricFrame."
+
+_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR = "Ambiguous Error Metric. Please only provide one of: error_bars and conf_intervals."
+
+_METRICS_NOT_LIST_OR_STR_ERROR = "Metric should be a string of a single metric or a list of the metrics in provided MetricFrame, but {0} was provided"
+_NEG_ERROR_OR_FLIPPED_BOUNDS_ERROR = "upper_error, lower_error, and symmetric_error must be positive and upper_bound must be greater than lower_bound"
+_INVALID_ERROR_MAPPING = "Invalid error mapping, each metric must have all the key-value pairs that make up an Error Format"
+
+def plot_metric_frame(metric_frame: MetricFrame,
+                     plot_type: str="scatter",
+                     metrics: Union[List[str], str]=None,
+                     error_bars: Union[List[str], str]=None,
+                     conf_intervals: Union[List[str], str]=None,
+                     axs=None,
+                     show_plot=True,
+                     plot_error_labels: bool=True,
+                     text_precision_digits: int=4,
+                     text_fontsize: int=8,
+                     text_color: str="black",
+                     text_ha: str="center",
+                     capsize=10,
+                     colormap="Pastel1",
+                     subplot_shape=None,
+                     figsize=None
+                     ):
+    """Visualization for metrics with statistical error bounds.
+
+    Plots a given metric and its given error (as described by the `error_mapping`)
+
+    This function takes in a :class:`fairlearn.metrics.MetricFrame` with precomputed metrics and metric errors
+    and a `error_mapping` dict to interpret the columns (??) of the `MetricFrame`.
+    
+    Parameters
+    ----------
+    metric_frame : fairlearn.metrics.MetricFrame
+        The collection fo disaggregated metric values, along with the metric errors.
+
+    plot_type : str
+        The type of plot to display. i.e. "bar", "line", etc
+
+    metrics : str or list of str
+        The name of the metrics to plot. Should match columns from the given `MetricFrame`
+
+    axs : matplotlib.axes._axes.Axes, optional
+        Custom Matplotlib axes to which this function can plot
+
+    show_plot : bool, optional
+        Whether to show the plot or not
+
+    plot_error_labels : bool, optional
+        Whether or not to plot numerical labels for the error bounds
+
+    text_precision_digits : int, optional
+        The number of digits of precision to show for error labels
+
+    text_font_size : int, optional
+        The font size to use for error labels
+
+    text_color : str, optional
+        The font color to use for error labels
+
+    text_ha : str, optional
+        The horizontal alignment modifier to use for error labels
+
+    Returns
+    -------
+    matplotlib.axes._axes.Axes
+    """
+    try:
+        import matplotlib.pyplot as plt
```

Comment:
I'm surprised that `flake8` isn't complaining about this - usually we put imports at the top of the file @romanlutz ?